### PR TITLE
[flutter_tools] write example of android SDK integration test

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

*Replace this paragraph with a description of what this PR is doing. If you're modifying existing behavior, describe the existing behavior, how this PR is changing it, and what motivated the change. If you're changing visual properties, consider including before/after screenshots (and runnable code snippets to reproduce them).*

## Related Issues

*Replace this paragraph with a list of issues related to this PR from our [issue database]. Indicate, which of these issues are resolved or fixed by this PR. There should be at least one issue listed here.*

## Tests

I added the following tests:

*Replace this with a list of the tests that you added as part of this PR. A change in behaviour with no test covering it
will likely get reverted accidentally sooner or later. PRs must include tests for all changed/updated/fixed behaviors. See [Test Coverage].*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I signed the [CLA].
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.
- [ ] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [ ] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by jonahwilliams** [comment]
I don't think we can run this on CI, but this test would have caught the doctor regression.

For discussion @christopherfujino @jmagman @zanderso

File: packages/flutter_tools/test/integration.shard/android_workflow_integration_test.dart [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+import 'package:platform/platform.dart';
+import 'package:archive/archive.dart';
+import 'package:process/process.dart';
+
+import '../src/common.dart';
+
+const String kAndroidRepo = 'https://dl.google.com/android/repository/';
+
+const Map<String, String> kPlatformTools = <String, String>{
+  'windows': 'commandlinetools-win-6200805_latest.zip',
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/integration.shard/android_workflow_integration_test.dart:21]: not sure how stable the URL is .. it has a `latest` in there

**Comment by zanderso** [comment]
Is there some way we could decouple the download and extraction from the test? That is, have a manual-ish process for updating and extracting the latest SDK, which will then be picked up by the CI test from a static location, or swaped to a new static location by a small PR?

Not sure if that makes sense, just trying to think about how to automate as much as possible to get something going.

**Comment by jonahwilliams** [comment]
I think we'd need it to be automated somehow, assuming the list of tests like this keep growing. Re-uploading somewhere doesn't help for Cirrus since downloading to those machines is slow, it would need to be persistent across test runs. I'm not sure if LUCI can do that. cc @digiter @godofredoc is this something that the devicelab could automate?

**Comment by digiter** [comment]
The flutter/engine recipe installs Xcode on demand from a CIPD path. We specific the Xcode version at https://github.com/flutter/infra/blob/5348bfc422fda0a756611843c211259e15d06ec7/config/main.star#L340. I guess you want something similar.

File: packages/flutter_tools/test/integration.shard/android_workflow_integration_test.dart (hunk 1 above)
Comment (dnfield) [reply packages/flutter_tools/test/integration.shard/android_workflow_integration_test.dart:21]: I believe that version number gets incremented from time to time, and that file may disappear if the version is old enough.

**Comment by dnfield** [comment]
Once upon a time, we packaged up an Android SDK into a cipd package for use like this. It was painful to upgrade and maintain, and gradle learned how to do grab the deps that are needed anyway.

**Comment by jonahwilliams** [comment]
Closing this for now

**Pull request closed without merging**